Sources/SnapshotTesting/Snapshotting/UIImage.swift

```diff
@@ -169,83 +169,77 @@ private func diff(_ old: UIImage, _ new: UIImage) -> UIImage {
 #endif
 
 #if os(iOS) || os(tvOS) || os(macOS)
-import CoreImage.CIKernel
-import MetalPerformanceShaders
```

Comment:
If there is a way to know if Metal is supported could we keep the previous implementation in case it is more performant than the alternative?

Reply:
Also if only metal backport of the threshold filter is the issue why to remove the use of all core image filters rather than using CIColorThreshold for iOS 14+ and fallback to CPU otherwise?

Reply:
You can check metal support like so: 

```swift
MTLCreateSystemDefaultDevice() == nil
```